src/main/java/com/marklogic/appdeployer/command/AbstractResourceCommand.java

```diff
@@ -30,6 +26,7 @@
 
     @Override
     public void execute(CommandContext context) {
+		setIncrementalMode(context.getAppConfig().getIncrementalDeploy());
```

Comment:
I see the problem now for roles - since DeployRolesCommand calls super.execute, this will get set back to "true" for the second deployment. 

How about if we do this - we make a "protected boolean isIncrementalDeployEnable(CommandContext)" method, and that defaults to "return context.getAppConfig().isIncrementalDeploy()". 

DeployRolesCommand can then override this method and say - if I'm doing the second pass, I'll always return false here.